sdk/core/azure-core/src/main/java/com/azure/core/http/policy/ProtocolPolicy.java

```diff
@@ -40,7 +40,8 @@ public ProtocolPolicy(String protocol, boolean overwrite) {
             try {
                 context.getHttpRequest().setUrl(urlBuilder.setScheme(protocol).toURL());
             } catch (MalformedURLException e) {
-                return Mono.error(e);
+                return Mono.error(new RuntimeException(
+                    String.format("Failed to set protocol %s to http request.", protocol), e));
```

Comment:
Similar comments for this and other policy error messages.

Reply:
Done.